c7n/filters/core.py

```diff
@@ -971,3 +974,64 @@ def reorder(self, items, key=None):
             return items[::-1]
         else:
             return sorted(items, key=key, reverse=(self.order == 'desc'))
+
+
+class CELFilter(Filter):
```

Comment:
can we do a pr with this just registered conditionally on celpy availability as a library first a simple try/except around import along with a global filter registry should suffice to get going.

Reply:
Just pushed up this conditional logic.

This most recent commit doesn't have `cel` added to the global filter registry, though I created `cel` filter classes for both Lambdas and EC2s. My current example for Lambdas directly uses this underlying `CELFilter`, while for EC2s I added logic in `process()` to prefetch the images, which allows for more EC2-specific filtering capabilities. If we add `cel` into the global filter registry, these resource-level implementations will run into circular import issues.

Do you want resource-specific additions to be separate from this PR? If so, I can pull this logic out and add `cel` into the global filter registry.

Reply:
This will run if you want to locally install https://github.com/cloud-custodian/cel-python/pull/19 and test it. Need to work on getting cel-python published on PyPI